airflow/providers/yandex/secrets/secrets_manager.py

```diff
@@ -0,0 +1,209 @@
+    the path ``airflow/config/sql_alchemy_conn``, the config with key ``sql_alchemy_conn`` would be
+    accessible.
+
+    When the prefix is empty, keys will use the Lockbox Secrets without any prefix.
+
+    .. code-block:: ini
+
+        [secrets]
+        backend = airflow.providers.yandex.secrets.secrets_manager.SecretsManagerBackend
+        backend_kwargs = {"connection_id": "<connection_ID>", "folder_id": "<folder_ID>"}
+
+    You need to specify id of yandexcloud connection to connect to Yandex Lockbox with.
+    Also, you need to specify the Yandex Cloud folder ID to search for Yandex Lockbox secrets in.
+
+    :param connection_id: Specifies the connection ID to connect to Yandex Lockbox with.
+        If set to None (null), requests will use default connection_id from YandexCloudBaseHook.
+        Default: "yandexcloud_default"
+    :param folder_id: Specifies the folder ID to search for Yandex Lockbox secrets in.
+        If set to None (null), requests will use the connection folder_id.
+    :param connections_prefix: Specifies the prefix of the secret to read to get Connections.
+        If set to None (null), requests for connections will not be sent to Yandex Lockbox.
+        Default: "airflow/connections"
+    :param variables_prefix: Specifies the prefix of the secret to read to get Variables.
+        If set to None (null), requests for variables will not be sent to Yandex Lockbox.
+        Default: "airflow/variables"
+    :param config_prefix: Specifies the prefix of the secret to read to get Configurations.
+        If set to None (null), requests for variables will not be sent to Yandex Lockbox.
+        Default: "airflow/config"
+    :param sep: Specifies the separator used to concatenate secret_prefix and secret_id.
+        Default: "/"
+    """
+
+    def __init__(
+        self,
+        connection_id: str | None = None,
+        folder_id: str | None = None,
+        connections_prefix: str | None = "airflow/connections",
+        variables_prefix: str | None = "airflow/variables",
+        config_prefix: str | None = "airflow/config",
+        sep: str = "/",
+    ):
+        super().__init__()
+
+        self.connection_id = connection_id
+        self.folder_id = folder_id
+        self.connections_prefix = connections_prefix.rstrip(sep) if connections_prefix is not None else None
+        self.variables_prefix = variables_prefix.rstrip(sep) if variables_prefix is not None else None
+        self.config_prefix = config_prefix.rstrip(sep) if config_prefix is not None else None
+        self.sep = sep
+
+    @property
+    def _client(self):
+        """Create a Yandex Cloud SDK client."""
+        return self._base_hook.client
+
+    @cached_property
+    def _base_hook(self) -> YandexCloudBaseHook:
+        return YandexCloudBaseHook(
+            yandex_conn_id=self.connection_id,
+            default_folder_id=self.folder_id,
+        )
```

Comment:
I think this one required to have a connection id somewhere else, e.g. Environment Variable or Airflow DB Backend first. All other providers provide ability to use it entirely without other backend. 

Better to have an ability to provide credentials thought backend_config, by the same way as it done in other community providers secrets backends:

**SystemsManagerParameterStoreBackend from Amazon Provider**
https://github.com/apache/airflow/blob/71c726d52d5a8a30f59268cc175560a4244c8016/airflow/providers/amazon/aws/secrets/systems_manager.py#L64-L67

**VaultBackend from Hashicorp provider**
https://github.com/apache/airflow/blob/71c726d52d5a8a30f59268cc175560a4244c8016/airflow/providers/hashicorp/secrets/vault.py#L66-L85

**CloudSecretManagerBackend from Google Provider**
https://github.com/apache/airflow/blob/71c726d52d5a8a30f59268cc175560a4244c8016/airflow/providers/google/cloud/secrets/secret_manager.py#L75-L79

**AzureKeyVaultBackend from Microsoft Azure provider**
https://github.com/apache/airflow/blob/71c726d52d5a8a30f59268cc175560a4244c8016/airflow/providers/microsoft/azure/secrets/key_vault.py#L83-L91